File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> ans;
        unordered_map<string,vector<string>> mp;
        for(int i=0;i<strs.size();i++)
        {
            string original=strs[i]; 
            string newstring=original;
            sort(newstring.begin(),newstring.end());
            mp[newstring].push_back(original);
        }
        for(auto i:mp)
        {
            ans.push_back(i.second);
        }
        return ans;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String,ArrayList<String>> mp=new HashMap(); 
        List<List<String>> ans=new ArrayList<>(); 
        for(int i=0;i<strs.length;i++)
        {
            char arr[]=strs[i].toCharArray();
            Arrays.sort(arr); 
            String str=new String(arr);
            if(!mp.containsKey(str))
            {
                mp.put(str,new ArrayList<>());
            }
            mp.get(str).add(strs[i]);
        }
        ans.addAll(mp.values());
        return ans;
    }
}